types/next/app.d.ts

```diff
@@ -0,0 +1,17 @@
+import * as React from "react";
+import { NextContext } from ".";
+
+export interface AppComponentProps {
+    Component: React.ComponentType<any>;
+    pageProps: any;
+}
+
+export interface AppComponentContext {
+    Component: any;
+    router: any; // TODO: could be SingletonRouter?
```

Comment:
Any spotted here too

Reply:
yeah. I'm planning to use `SingletonRouter` for this property but not sure if it's the right type for it.

Reply:
It is